### DISCUSSION TITLE
blog/ebpf-sidecar-and-service-mesh/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/rootsongjc.github.io#302] [Announcements]
# blog/ebpf-sidecar-and-service-mesh/

不管有没有 eBPF，在可预见的未来，服务网格都会基于运行在用户空间的 sidecar 代理（proxyless 模式除外）。

https://jimmysong.io/blog/ebpf-sidecar-and-service-mesh/

**Comment by 0xleizhang** [comment]
光抱怨不行怎么能行，Cilium service mesh 不就来了吗，人家已经干出来了，可以做Layer7层的转发。
https://www.cncf.io/blog/2022/07/20/cilium-1-12-ga-cilium-service-mesh-and-other-major-new-features-for-enterprise-kubernetes/

**Comment by icefed** [comment]
我和博主持相反的观点，认为 epbf 的 per-node 模式才是未来的方向。并且觉得博主认为 per-node 的一些缺点有些牵强。
1、代理的资源消耗难道不是随着 POD 的数量增长而增长吗？我们这边因为 istio 每个 sidecar 占用近百兆的内存而放弃了 istio。
2、爆炸半径，这个我表示认同，per-node 确实有中心化的缺点。
3、代理升级，我认为 per-node 更方便呢。难道你为了升级代理给每个应用做个升级？
4、安全边界，我认为两者相差不多。

然后提一个 sidecar 无法支持的情况，job 类型无法使用 sidecar 或者无法很优雅的使用，并且 Kubernetes sidecar container proposal 已经被否了。可能博主的使用场景中很少使用 job，但是不能忽略。

**Reply by SaberFateLover** [reply]
楼主您好，想咨询您两个问题：
1、目前打算引入sidecar ,想问一下楼主，sidecar的内存占用近百兆这边有调研具体原因吗。
2、istio这边有做系统性的压测吗，效果怎么样。

**Reply by icefed** [reply]
1、没有找具体原因，因为使用中发现不是个别情况，是所有的 sidecar 占用都很高。
2、我们主要是使用 mtls，对请求并发要求不高，没有针对性压测。

**Comment by SaberFateLover** [comment]
蚂蚁已经在做sidecar 的下沉工作了，可见per-host还是有很大优势，目前基于iptable网络链路确实很长，基于eBPF的实现是大势所趋。
